### ISSUE TITLE
sphinx-lint does not detect inline emphasis glued to inline literal.

### ISSUE THREAD

**Issue by JulienPalard** [sphinx-contrib/sphinx-lint#52]
As mentionned in https://github.com/python/python-docs-es/issues/2200#issuecomment-1320770780:

``` some words *with stars*``backtick-words`` ```

is detected as valid by sphinx-lint, while it is not:

```
$ rst2pseudoxml tests/fixtures/xfail/inline-literal-glued-to-emphasis.rst 
tests/fixtures/xfail/inline-literal-glued-to-emphasis.rst:1: (WARNING/2) Inline emphasis start-string without end-string.
<document source="tests/fixtures/xfail/inline-literal-glued-to-emphasis.rst">
    <paragraph>
        some words 
        <problematic ids="id2" refid="id1">
            *
        with stars*``backtick-words``.
    <system_message backrefs="id2" ids="id1" level="2" line="1" source="tests/fixtures/xfail/inline-literal-glued-to-emphasis.rst" type="WARNING">
        <paragraph>
            Inline emphasis start-string without end-string.
```

But it is detected by docutils so it's not a big issue.
